### PR TITLE
Fixes to make test-utils to work in external apps

### PR DESCRIPTION
**Pull request by ibgreen**
For #1423
#### Background
- Fixes to get test-utils working in other apps, using deck.gl 5.1
#### Change List
- Remove waitForFunction that was moved to probe.gl
- Add timer to ensure puppet function is available.
- Add temporary support for `viewports` to support 5.1

### CODE REVIEW

File: src/test-utils/src/render-test.js [hunk 1]
```diff
@@ -117,8 +117,14 @@ export default class RenderTest {
...
   _onComplete() {
     this.reportFinalResult({passed: this.passed, renderTest: this});
+    this._reportToTestDriver();
+  }
+
+  _reportToTestDriver(firstTime = false) {
     if (window.renderTestComplete) {
       window.renderTestComplete(JSON.stringify(this.passed));
+    } else {
+      window.setTimeout(this._reportToTestDriver.bind(this, false), 1000);
```
Comment (jianhuang01) [inline src/test-utils/src/render-test.js]: what's the purpose of this line?

File: src/test-utils/src/drivers/render-test-driver.js [hunk 2]
```diff
@@ -35,21 +35,14 @@ const SERVER_CONFIG = {
...
-
   run(config = SERVER_CONFIG) {
     this.console.log(addColor('Running rendering tests in Chrome instance...', COLOR.YELLOW));
     this.time = Date.now();
     return Promise.resolve()
       .then(_ => this.startServer(config))
       .then(_ => this.startBrowser())
       .then(_ => this.newPage())
-      .then(_ => this.waitForFunction('renderTestComplete'))
+      .then(_ => this.waitForBrowserMessage('renderTestComplete'))
```
Comment (jianhuang01) [inline src/test-utils/src/drivers/render-test-driver.js]: where do we send out 'renderTestComplete' message?

File: src/test-utils/src/render-test.js (hunk 1 above)
Comment (ibgreen) [reply src/test-utils/src/render-test.js]: Added comment.

File: src/test-utils/src/drivers/render-test-driver.js (hunk 2 above)
Comment (ibgreen) [reply src/test-utils/src/drivers/render-test-driver.js]: Just above your other comment. `renderTestComplete` becomes available as `window.renderTestComplete`

**Comment by georgios-uber** [comment]
This breaks yarn.lock, making a fix now.

**Comment by georgios-uber** [comment]
https://github.com/uber/deck.gl/pull/1509

**Pull request merged**